Source/CarthageKit/Project.swift

```diff
@@ -913,9 +913,11 @@ public final class Project { // swiftlint:disable:this type_body_length
 						continue
 					}
 
-					return Result(at: dependencyCheckoutURL, attempt: {
+					if let error = Result(at: dependencyCheckoutURL, attempt: {
```

Comment:
It seems to me that restoring the original structure will be more readable / understandable:

```swift
do {
	try fileManager.createSymbolicLink(atPath: dependencyCheckoutURL.path, withDestinationPath: linkDestinationPath)
} catch let error as NSError {
	return .failure(.writeFailed(dependencyCheckoutURL, error))
}
```